### PR TITLE
Decorate a Class

### PR DESCRIPTION
**Pull request by Kingno3l**
### These are the codes implemented in this branch;

**Created a class Nameable.**

- [ ] Implemented a method called correct_name that will raise a NotImplementedError.Person class inherits from Nameable.

**Turned my Person class to Nameable**
- [ ] Person class inherits from Nameable
- [ ] This class has a method correct_name implemented. It simply returns the name attribute.

**Prepared base Decorator**

- [ ] It inherits from Nameable.
- [ ] In the constructor, I assigned a nameable object from params to an instance variable.
- [ ] Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

**Prepared CapitalizeDecorator and TrimmerDecorator**

- For the CapitalizeDecorator:

- [ ] Created a class that inherits from the base Decorator class.
- [ ] Implemented a method correct_name that capitalizes the output of @nameable.correct_name.

- For the TrimmerDecorator:

- [ ] Created a class that inherits from the base Decorator class.
- [ ] Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @Kingno3l,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly add the Future Features section and menu link to your readme file

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by julie-ify** [comment]
@Kingno3l, you have not made the correct change to the readme file. Your menu section should have the Future Features link on the menu
image
and the futures features section itself. Remember to remove the section description
image

**Review by julie-ify — COMMENTED** [review]
Hi @Kingno3l,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly add the future features correctly.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
+
+class Person < Nameable
   attr_accessor :id, :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @age = age
     @id = generate_id
     @name = name
     @age = age
```
Comment (OyePriscilla) [inline person.rb]: - [ ] Please kindly take note that @age instance was declared twice, I think you can remove one for proper execution of your app. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Kingno3l 👏 👏 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implement composition (as an example of the design pattern). ✔️ 
Run a program using the command line. ✔️ 

## Required Changes ♻️

- [ ] Please kindly ensure that the link you added to the Ruby stack (client section of the README file) is accessible by the general public. I got the error message below 👇 

Untitled


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,17 +1,24 @@
-class Person
+require_relative 'Nameable'
```
Comment (FrederickMih) [inline person.rb]: - Please kindly refer to the correct file name which is **nameable**. This issue produces an error when running this file. Let's kindly fix :wrench: it before merging our PR

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @Kingno3l, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Person's name have been decorated as expected :+1: :rocket: 
Screenshot from 2023-11-21 23-23-03


## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (Kingno3l) [reply person.rb]: Done sir...
@FrederickMih

**Pull request merged**